lib/XTR/ScheduleReminderTask.php

```diff
@@ -0,0 +1,192 @@
+                        case ScheduleReminder::$OPTION_BEFORE_START:
+                            $typeOptionText = 'starting';
+                            break;
+                        case ScheduleReminder::$OPTION_AFTER_START:
+                            $typeOptionText = 'started';
+                            break;
+                        case ScheduleReminder::$OPTION_BEFORE_END:
+                            $typeOptionText = 'ending';
+                            break;
+                        case ScheduleReminder::$OPTION_AFTER_END:
+                            $typeOptionText = 'ended';
+                            break;
+                        default:
+                            throw new \Xibo\Exception\NotFoundException('Unknown option');
+                    }
+
+                    // Create a notification
+                    $subject = sprintf(__("Reminder for %s"), $title);
+                    if ($reminder->option == ScheduleReminder::$OPTION_BEFORE_START || $reminder->option == ScheduleReminder::$OPTION_BEFORE_END) {
+                        $body = sprintf(__("The event (%s) is %s in %d %s"), $title, $typeOptionText, $reminder->value, $typeText);
+                    } elseif ($reminder->option == ScheduleReminder::$OPTION_AFTER_START || $reminder->option == ScheduleReminder::$OPTION_AFTER_END) {
+                        $body = sprintf(__("The event (%s) has %s %d %s ago"), $title, $typeOptionText, $reminder->value, $typeText);
+                    }
+
+                    // Send a notification to the event owner
+                    $notification = $this->notificationFactory->createEmpty();
+                    $notification->subject = $subject;
+                    $notification->body = $body;
+                    $notification->createdDt = $this->date->getLocalDate(null, 'U');
+                    $notification->releaseDt = $reminder->reminderDt;
+                    $notification->isEmail = $reminder->isEmail;
+                    $notification->isInterrupt = 0;
+                    $notification->userId = $schedule->userId; // event owner
+
+                    // Send
+                    $notification->save();
+
+                    // Is this schedule a recurring event?
+                    if ($schedule->recurrenceType != '') {
+
+                        $now = $this->date->parse();
+                        $remindSeconds = $reminder->value * $type;
+                        $nextReminderDate = $schedule->getNextReminderDate($now, $reminder, $remindSeconds);
+                        if($nextReminderDate != null) {
+
+                            $reminder->reminderDt = $nextReminderDate;
+                            $reminder->save();
+                            $this->appendRunMessage(__('Reminder added for  '. $title));
+                        } else {
+
+                            // Remove the reminder
+                            $reminder->delete();
```

Comment:
Should we be removing reminders that have passed? Seems like their reminder dates have gone by and we can safely just ignore them now?